tests/test_aging.c

```diff
@@ -0,0 +1,567 @@
+    }
+    sr_session_switch_ds(data->sess, SR_DS_RUNNING);
+    return 0;
+}
+
+int
+find_file(void **state, const char *search, int *found)
+{
+    DIR *d;
+    struct dirent *dir;
+    test_data_t *data = (test_data_t *) (*state);
+    char *file_name = NULL;
+
+    d = opendir(data->a_path);
+    if (!d) {
+        fprintf(stderr, "Failed to open the directory\n");
+        return EXIT_FAILURE;
+    }
+
+    if (asprintf(&file_name, "%s", search) == -1) {
+        fprintf(stderr, "Asprintf failed\n");
+    }
+
+    /** Read whole directory **/
+    while ((dir = readdir(d))) {
+
+        /** Skip current and parent directories **/
+        if (dir->d_name[0] == '.') {
+            continue;
+        }
+
+        /** Find a file that should've been archived **/
+        if (strlen(dir->d_name) == strlen(file_name)) {
+            if (!strcmp(dir->d_name, file_name)) {
+                *found = 1;
+                break;
+            }
+        }
+    }
+    free(file_name);
+    closedir(d);
+    return 0;
+}
+
+static int
+teardown(void **state)
+{
+    int rc = 0, ret;
+    test_data_t *data = (test_data_t *) (*state);
+
+    if (data) {
+
+        /** Kill forked process **/
+        if (data->pid) {
+            kill(data->pid, SIGINT);
+            waitpid(data->pid, &ret, 0);
+            if (WIFSIGNALED(ret)) {
+                fprintf(stderr, "Child has been terminated by a signal no: %d\n", WTERMSIG(ret));
+            }
+        }
+
+        /** Remove notif and archive directories **/
+        rc = remove_dir(data->n_path, NULL);
+        rc = remove_dir(data->a_path, NULL);
+
+        /** Free all **/
+        sr_disconnect(data->conn);
+        free(data->pidfile);
+        free(data->path_to_file);
+        free(data->n_path);
+        free(data->a_path);
+        free(data);
+    }
+    return rc;
+}
+
+static int
+setup(void **state)
+{
+    int rc = 1;
+    time_t start;
+    test_data_t *data = (test_data_t *)calloc(1, sizeof(test_data_t));
+    if (!data) {
+        fprintf(stderr, "Calloc failed\n");
+        goto cleanup;
+    }
+
+    /** Retrieve notif path and archive path **/
+    if ((data->n_path = get_test_path("notif/")) == NULL) {
+        goto cleanup;
+    }
+    if ((data->a_path = get_test_path("archiv/")) == NULL) {
+        goto cleanup;
+    }
+    data->len_npath = strlen(data->n_path);
+    data->len_apath = strlen(data->a_path);
+    
+    /** Create notif and archive directories **/
+    if ((rc = create_dir(data->n_path))) {
+        goto cleanup;
+    }
+    if ((rc = create_dir(data->a_path))) {
+        goto cleanup;
+    }
+
+    /** Store path to pidfile **/
+    if ((rc = asprintf(&data->pidfile, "%s%s", data->n_path, "my_pidfile")) == -1) {
+        goto cleanup;
+    }
+
+    /** Run archivation loop **/
+    data->pid = fork();
+    if (data->pid == 0) {
+        if (execl(SR_BINARY_DIR "/sysrepo-plugind", SR_BINARY_DIR "/sysrepo-plugind", "-d", "-p", data->pidfile, NULL)) {
```

Comment:
This line starts `sysrepo-plugind` which will immediately start loading whatever plugins which are installed on the users' machine. That's not the expected behavior of a unit test.